### PR TITLE
[Filesystem] use local PHP web server to test HTTP stream wrappers

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

same approach as #54360 (see https://github.com/symfony/symfony/actions/runs/8387183697/job/22968895866#step:8:2558 for a failure)

### CODE REVIEW

**Comment by xabbuh** [comment]
Status: Needs work

**Comment by xabbuh** [comment]
Status: Needs Review

File: src/Symfony/Component/Filesystem/Tests/FilesystemTest.php [hunk 1]
```diff
@@ -162,15 +164,23 @@ public function testCopyCreatesTargetDirectoryIfItDoesNotExist()
...
+        if (!\in_array('http', stream_get_wrappers())) {
+            $this->markTestSkipped('"http" stream wrapper is not enabled.');
         }
-        $sourceFilePath = 'https://symfony.com/images/common/logo/logo_symfony_header.png';
+
+        $finder = new PhpExecutableFinder();
+        $process = new Process(array_merge([$finder->find(false)], $finder->findArguments(), ['-dopcache.enable=0', '-dvariables_order=EGPCS', '-S', '127.0.0.1:8057']));
+        $process->setWorkingDirectory(__DIR__.'/Fixtures/web');
+
+        $process->start();
```
Comment (stof) [inline src/Symfony/Component/Filesystem/Tests/FilesystemTest.php:177]: shouldn't we stop the process in a finally  block to ensure it is properly closed ?

Comment (xabbuh) [reply src/Symfony/Component/Filesystem/Tests/FilesystemTest.php:177]: indeed, done

**Review by nicolas-grekas — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Pull request merged**